browser/views.py

```diff
@@ -713,6 +713,65 @@ def delete_investigator_record(request, person_id, research_id):
     return HttpResponseRedirect(reverse('person', args=(person_id,)))
 
 
+@staff_member_required
+def edit_affiliation(request, person_id, affiliation_id):
+    person = get_object_or_404(Person, pk=person_id)
+    affiliation = get_object_or_404(Affiliation, pk=affiliation_id)
+    template = "browser/affiliation_edit.html"
+    institution = None
+
+    form = EditAffiliationForm(initial={'institution': affiliation.institution,
+                                        'position': affiliation.position,
+                                        'institution_id': affiliation.institution.id if affiliation.institution else None})
+    context = {
+        'form': form,
+        'person': person
+    }
+
+    if request.method == 'POST':
+        form = EditAffiliationForm(request.POST, instance=person)
+        check_handle_affiliation = handle_institution(form, None, request)
+        if check_handle_affiliation['status']:
+            Affiliation.objects.filter(id=affiliation_id).update(position=form.cleaned_data.get('position'),
+                                                     institution=check_handle_affiliation['institution'])
+            return HttpResponseRedirect(reverse('person', args=(person_id,)))
+        else:
+            messages.add_message(request, messages.ERROR,
+                                 'The above institute does not exist in the database. '
+                                 'Please check the checkbox below to create new institute')
+            context.update({'form': form})
+            return render(request, template, context)
+
+    return render(request, template, context)
+
+
+@staff_member_required
+def delete_affiliation_record(request, person_id, affiliation_id):
+    if request.method == 'POST':
+        Affiliation.objects.filter(id=affiliation_id).delete()
+
+    return HttpResponseRedirect(reverse('person', args=(person_id,)))
+
+
+# This helper checks if an institution is to be created or an existing institution is to be retrieved
+def handle_institution(form, institution, request):
+    if form.is_valid():
+        if form.cleaned_data.get('institution'):
+            # If institution_id is not populated then create the institution before updating affiliation
+            if not form.cleaned_data.get('institution_id'):
+                # Checking if create institution radio button has been checked
+                if not form.cleaned_data.get('create_institution'):
+                    return {'status': False, 'institution': institution}
+                else:
+                    institution = Institution.objects.create(
+                        name=form.cleaned_data.get('institution'),
+                        changed_by=request.user
+                    )
+            else:
+                institution = Institution.objects.get(id=form.cleaned_data.get('institution_id'))
+        return {'status': True, 'institution': institution}
```

Comment:
why is this called 'status' if its value is only False or True?

Reply:
just return a tuple and put the return value into two variables when calling the function